src/basic/process-util.c

```diff
@@ -855,8 +855,11 @@ noreturn void freeze(void) {
 
         sync();
 
-        for (;;)
-                pause();
+        for (;;) {
+                siginfo_t si = {};
+
+                (void) waitid(P_ALL, 0, &si, WEXITED);
```

Comment:
I think we should be more defensive here, after all we call this when things are seriously fucked up already. Specifically I think we should add some minimal error handling. i.e. instead of looping around waitid() even if it fails I think we should add one more safety net:

```c
for (;;) {
        siginfo_t si = {};

        if (waitid(…) < 0 && errno != EINTR)
                break;
}

for (;;) /* extra safety net, in case waitid() fails for some unexpected reason */
        pause();
```